src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -648,6 +648,59 @@ void MPS::apply_diagonal_matrix(const AER::reg_t &qubits, const cvector_t &vmat)
   apply_matrix(qubits, diag_mat);
 }
 
+void MPS::apply_kraus(const reg_t &qubits,
+                   const std::vector<cmatrix_t> &kmats,
+                   RngEngine &rng) {
+  reg_t internal_qubits = get_internal_qubits(qubits);
+  apply_kraus_internal(qubits, kmats, rng);
+
+}
+void MPS::apply_kraus_internal(const reg_t &qubits,
```

Comment:
I think this would be clearer with just a single function that converts the args to the internal qubits and uses them:

```cpp
void MPS::apply_kraus(const reg_t &qubits,
                   const std::vector<cmatrix_t> &kmats,
                   RngEngine &rng) {

   reg_t internal_qubits = get_internal_qubits(qubits);
  // code from apply_kraus_internal using internal qubits

}
```

Reply:
I can, of course unite the functions, but I would rather not, because of the following reasons:
1. This is the structure that is defined for all the operations - a short api function that only gets the internal qubits and the internal function that actually does something. There is a logical separation between the 2 levels.
2. My initial bug came from not doing the separation, and forgetting that it was necessary. That is also the reason I moved the functions to be inside matrix_product_state_internal.[ch]pp.
3. Sometimes I may want to access the internal function directly. For example, when I call it from another internal function.